bee-ledger/src/types/receipt.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::types::{error::Error, TreasuryOutput};
+
+use bee_common::packable::{Packable, Read, Write};
+use bee_message::{
+    constants::IOTA_SUPPLY,
+    input::Input,
+    milestone::MilestoneIndex,
+    output::Output,
+    payload::{receipt::ReceiptPayload, Payload},
+};
+
+/// A type that wraps a receipt and the index of the milestone in which it was included.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Receipt {
+    inner: ReceiptPayload,
```

Comment:
consider renaming `inner` to `payload` or `receipt_payload`

Reply:
I don't really like payload as it gives the idea that it's an actual `Payload` type and then can be generic. I like `inner` because `Receipt` is just a wrapper type.